src/lib/middleware/userPreferencesMiddleware.ts

```diff
@@ -0,0 +1,35 @@
+import { NextFunction } from "express"
+import { ArtsyRequest, ArtsyResponse } from "./artsyExpress"
+import request from "superagent"
+import { API_URL } from "config"
+
+const DEFAULT_METRIC = "cm"
+
+export const userPreferencesMiddleware = async (
+  req: ArtsyRequest,
+  res: ArtsyResponse,
+  next: NextFunction
+) => {
+  if (!!req.user) {
+    let metric
+
+    try {
+      const endpoint = `${API_URL}/api/v1/me`
```

Comment:
Can we use GraphQL for this, rather than relying on our gravity endpoint? Then we'll get proper generated types. Here's [a good example](https://github.com/artsy/force/blob/main/src/v2/Apps/RSS/rssServerApp.tsx#L19-L27) of how that can be done.

Reply:
(Just a non-blocking suggestion)

Reply:
Thank you for the example, I will take a look at it 🙏

Reply:
Done